feat: APIs for native integration with AJAX / SPAs / ReactJS / NextJS / ...

aeneasr opened the pull request:
## Proposed changes

This PR adds long-awaited capabilities for handling AJAX requests in the browser flows!

Closes #1138
Closes #668

sbussard commented on test/e2e/cypress/support/commands.js:
```diff
@@ -144,11 +146,9 @@ Cypress.Commands.add(
       })
-      .then((res) => {
+      .then(({ body }) => {
         console.log('Registration sequence completed: ', { email, password })
```
```suggestion
```

aeneasr commented:
All APIs have now been migrated. We still need to document the recovery and verification flows in the docs and also fix any failing e2e tests.

frederikhors commented:
Is this online? I mean, the docs?

frederikhors commented:
> Is this online? I mean, the docs?

Yes in the next version. Sorry.

frederikhors commented:
@aeneasr @vinckr can you please release this in Docker? The latest is `0.6.0-alpha.1` which doesn't contain this, right?

aeneasr commented:
While I understand that a date of delivery is helpful when planning, we stopped giving out due dates for features or milestones. It puts maintainers in an unfair spot as

1. there are many things going on in the community which need time (reviewing PRs, addressing issues, answering questions) and demand is no not easily foreseeable;
2. maintainers do a lot of things in their free time and can not commit to deliver something;
3. internal priorization can shift at any point in time

When we do give out due dates, we often encounter people who are pissed off because something was promised and not delivered "on time" (as free software ;) ). Unfortunately that is the sad truth of many interactions in this context which is why we stopped doing it altogether.

Having said that, you are in the privileged position of being part in a global open source community! If you think that you can contribute towards the particular feature, maintainers will do their best to give you the right pointers and review PRs.

If that's not an option, you can consider to become a Ory Sponsor on [Open Collective](https://opencollective.com/ory) or [Patreon](https://www.patreon.com/_ory). This helps us to employ more maintainers and technical staff - increasing velocity in development, community, interaction, and other areas! All collected money goes directly into this.

frederikhors commented:
I meant that it would be great to be your alpha/beta-tester of the news in the `master` branch if only we had an automatic build on each commit for a new version of images on the Docker Hub.

aeneasr commented:
Ah I see - like nightly builds? The problem with those though is that they override the latest tags which can often lead to issues :/

frederikhors commented:
> like nightly builds?

Yep! 😄

> override the latest tags

One should never use the latest tag unless they want to live with shivers down their spine (like me).

aeneasr commented:
Slept over it, we can't really do that because sometimes things on master change between revisions because we merge something broken, for example. Releases go through more QA. You can always build the images yourself with `make docker` which is also documented in the README: https://github.com/ory/kratos#build-docker

frederikhors commented:
Yeah. I can do that.

It's just a lot more complicated being a longer job, especially to test the image in various projects and environments.

> because we merge something broke

This is not a problem. The build fails. The latest valid build on Docker hub is the latest valid (not broken) commit.

aeneasr commented:
> because we merge something broken

such as SQL migrations which need to be corrected in the next version, leaving people relying on nightly builds with a broken database schema. Please rely on `make docker`

frederikhors commented:
> leaving people relying on nightly builds with a broken database schema

This is expected if I use `latest` tag.

Anyway ok, don't worry.
